### ISSUE TITLE
DLND-Week 1

### ISSUE THREAD

**Issue by chenghong-lin-nu** [Hclmaster/blog#1]
## Overview
![image](https://user-images.githubusercontent.com/16880879/35261692-5dd4867a-004c-11e8-8644-47001a7bbf9d.png)
* 需要对以下内容有所了解：
* [scikit-learn: Python机器学习库](http://scikit-learn.org/stable/)
* [Perceptron](https://en.wikipedia.org/wiki/Perceptron)
* [Gradient Descent](梯度下降)
* Numpy

## Anaconda
* Anaconda 能让你在数据科学的工作中轻松安装经常使用的程序包。
* 你还将使用它创建虚拟环境，以便更轻松地处理多个项目。
* Anaconda 简化了工作流程，并且解决了多个包和 Python 版本之间遇到的大量问题。

## Jupyter Notebook
* Jupyter notebook 是一种 Web 应用，能让用户将说明文本、数学方程、代码和可视化内容全部组合到一个易于共享的文档中。
* 以及一些jupyter notebook的快捷键（之后补完）

## Machine Learning
![image](https://user-images.githubusercontent.com/16880879/35275046-60e8db86-0079-11e8-8b51-ba1e72e89792.png)
* Machine Learning被分成了3个部分
![image](https://user-images.githubusercontent.com/16880879/35275245-163009ba-007a-11e8-9252-a7502b17df25.png)

## 线性回归模型
* 线性回归最适用于线性数据
* 线性回归会根据训练数据生成直线模型。
* 线性回归的目标是求取对训练数据而言的 “最优拟合” 直线。

e.g
```python
import pandas as pd
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt

bmi_life_data = pd.read_csv('bmi_and_life_expectancy.csv')
#print(dataframe)
y_value = bmi_life_data[['Life expectancy']]
x_value = bmi_life_data[['BMI']]

#print(z_value, y_value)
bmi_life_model  = LinearRegression()
bmi_life_model.fit(x_value, y_value)
laos_life_exp = bmi_life_model .predict([[21.07931]])
print(laos_life_exp)
```

### 多元线性回归
* 如果预测结果取决于多个变量，则需相应创建更加复杂的模型。只要所选自变量/预测变量适合当前场景，增加变量有助于改善预测结果。
* 增加预测变量，相当于增加图像维度。
* 若使用 n 个预测变量，那么模型公式则为：
![image](https://user-images.githubusercontent.com/16880879/35276897-f472160a-007f-11e8-8337-0e58e68c18d3.png)
